### ISSUE TITLE
fatal error: module 'airship_flutter' not found

### ISSUE THREAD

**Issue by andrewescutia** [urbanairship/airship-flutter#46]
# Preliminary Info

### What Airship dependencies are you using?
airship_flutter: ^3.0.1

### What are the versions of any relevant development tools you are using?
Flutter 1.12.13+hotfix.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f139b11009 (8 days ago) • 2020-03-30 13:57:30 -0700
Engine • revision af51afceb8
Tools • Dart 2.7.2

# Report

### What unexpected behavior are you seeing?
Unable to build to iOS device. receiving the following: fatal error: module 'airship_flutter' not found. I am able to build on the Beta Flutter channel but I'd rather be on Stable if possible.

### What is the expected behavior?
Build to iOS device

### What are the steps to reproduce the unexpected behavior?
* flutter create -i swift .
* follow the necessary migration steps here: https://flutter.dev/docs/development/ios-project-migration
* add airship_flutter: ^3.0.1 to pubspec.
* attempt to build to a device

### Do you have logging for the issue?
`Building App.framework for armv7...
    Building AOT snapshot in release mode (ios-release)...             14.3s
    Built to build/aot/.
    warning: parsing line table prologue at offset 0x6f697463 found unsupported version 0x00
    warning: line table parameters mismatch. Cannot emit.
    note: while processing /Users/aescutia/git/test_airship/build/aot/armv7/snapshot_assembly.o
    Project /Users/aescutia/git/test_airship built and packaged successfully.
    /Users/aescutia/git/test_airship/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'airship_flutter' not found
    @import airship_flutter;
     ~~~~~~~^~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description`

**Comment by marc-scig** [comment]
Thanks @andrewescutia, we'll take a look

**Comment by marc-scig** [comment]
Hey @andrewescutia,

I was able to reproduce your issue with a fresh app on Xcode 11.4, but following the migration instructions resolved it for me. What steps did you follow? The updating of linker flags, build scripts and embedded library references in particular all seem pretty crucial.

**Comment by andrewescutia** [comment]
Sorry, just got back to this. did you try a 'flutter build ios' that seems to be when I get it. I followed the migration steps in the linked doc. nothing was done other than that.

**Comment by marc-scig** [comment]
Hey @andrewescutia,

Sorry to leave you hanging like that. Are you still experiencing this issue?

**Comment by marc-scig** [comment]
It looks like one of the things that can cause this error is when there's a mismatch between the app's deployment target version (in Xcode under Build Settings -> iOS Deployment Target) and the platform version specified in the app's Podfile. The Airship SDK currently requires a minimum deployment target of 11.0, and you should see more informative CocoaPods related errors if the Podfile doesn't have this line uncommented:

`platform :ios, '11.0'`

However if the deployment target in Xcode is still set to a lower version, it looks like this will silently cause the Airship SDK pod not to be pulled in at build time, which will then result in those import errors further along. This isn't an issue when using `flutter run` or running the app in debug mode directly from Xcode, which is part of what makes it obscure.

So, long story short, if anyone is seeing this the first thing to check is whether the platform specified in the Podfile matches the deployment target in Xcode, and that they're both set to the correct minimum platform/target version (which is currently 11.0).

**Comment by andrewescutia** [comment]
that did it. I had to update min deployment target in Xcode, cant believe I missed that bit. Thanks!
